python/revng/ptml/actions.py

```diff
@@ -0,0 +1,79 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+from dataclasses import dataclass
+from typing import Callable, Dict, Optional, Tuple, Union
+
+from revng.support.location import Location
+
+
+@dataclass(frozen=True)
+class ActionMapping:
+    # Given a PipelineLocation, return the TupleTreePath for the model
+    rename: Optional[Callable[[Location], str]]
+    # Given a PipelineLocation, return the TupleTreePath for the model
+    comment: Optional[Callable[[Location], str]]
+
+
+def _model_path_generator(*args: Union[str, Tuple[int, bool]]) -> Callable[[Location], str]:
+    def generator(location: Location):
+        if len(args) == 0:
+            return ""
+
+        result = ""
+        for element in args:
+            if not result.endswith("::"):
+                result += "/"
+
+            if isinstance(element, str):
+                result += element
+            else:
+                if element[1]:
+                    # Path needs to be upcasted
+                    part = location.path_components[element[0]]
+                    kind = part.split("-")[-1]
+                    result += f"{part}/{kind}::"
+                else:
+                    result += location.path_components[element[0]]
+        return result
+
+    return generator
+
+
+def _common(*args: Union[str, Tuple[int, bool]]) -> ActionMapping:
+    return ActionMapping(
+        _model_path_generator(*args, "CustomName"),
+        _model_path_generator(*args, "Comment"),
+    )
+
+
+ACTION_MAPPING: Dict[str, ActionMapping] = {
```

Comment:
Document what are the `0` and `bool`s

Reply:
Added a nice and big docstring explaining what's going on